prefer youtube media URLs

redreceipt opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->
## ✏️ Description

<!--
Describe your changes, and why you're making them. Is this linked to an open issue, a Trello card, or another pull request? Link it here.
-->

Chooses Youtube links first for video

## 🔬 To Test
1. Look at Bayside devotional videos, should be youtube

## 📸 Screenshots
<!--
| Feature | Simulator | Figma | Links | Design Considerations |
| --- | --- | --- | --- | ---|
| Feature 1 | _attach image_ | _attach image_ | [Link to Figma](linkurl) | The reason that we chose to do... |
| Feature 2 | _attach image_ | _attach image_ | [Link to Figma](linkurl) | The reason that we chose to do... |
-->

<img width="565" alt="Screenshot 2023-02-02 at 10 30 35 AM" src="https://user-images.githubusercontent.com/2659478/216381563-2fa167c7-04c7-4c63-bd87-9965262c413c.png">


## ⚠️ To-do Before Merge

<!--
Include any notes about things that need to happen before this PR is merged, e.g.:
- [ ] Ensure PR #42 is merged
- [ ] Update Figma to reflect some design changes made in code
-->

## 📝 Checklist

<!--
This checklist is mostly useful as a reminder of small things that can easily be forgotten – it is meant as a helpful tool rather than hoops to jump through. 

Put an `x` in all the items that apply, make notes next to any that haven't been addressed, and remove any items that are not relevant to this PR.
-->

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)
- [x] For PR's with design updates/changes, I have consulted with the product designer and made all necessary adjustments in Figma

redreceipt commented on packages/apollos-ui-connected/src/ApollosPlayerConnected/index.js:
```diff
@@ -41,6 +41,7 @@ const ApollosPlayerConnected = ({
   Component,
   audioOnly,
+  mediaSources,
   mediaSource,
```
```suggestion
```

vinnyjth commented:
@redreceipt looking great - but there are some subtle bugs with existing code that will surface with these changes we might need to fix. 

These lines assume that the `videoId` and `videoDuration` come from the 0th video. We may need to apply the same selection strategy inside the `ApollosPlayerConnected` or (even better) consolidate some of this source/video selection to one of the components, maybe `ApollosPlayerConnected` shouldn't do any fetching on it's own? 

https://github.com/ApollosProject/apollos-apps/pull/2951/files#diff-dd8d85e421d3f1f939c3c28f8ebb6fb15ced20f0768a0c636d63559e0bd16892R51-R52

calebpanza commented:
This preference is gonna break tv... @conrad-vanl told me that we were in the process of removing all YouTube videos from Bayside. is that still happening?

redreceipt commented:
> This preference is gonna break tv... @conrad-vanl told me that we were in the process of removing all YouTube videos from Bayside. is that still happening?

TV will still use the native player

calebpanza commented:
That assumes we'll have both a native and YouTube file, though. Is that a safe assumption?

redreceipt commented:
@calebpanza I'll need to submit a PR to TV that filters out everything except HLS links

redreceipt commented:
ok @vinnyjth this is ready again. I removed the query from the `PlayerConnected` component (so it's not really connected anymore) and cleaned up the props to pass in all relevant data from the first time the query is run. All accompanying metadata comes from the preferred media now, YouTube or otherwise. I also cleaned up some unused props in the media player typescript components.
